launcher/ui/themes/ThemeManager.cpp

```diff
@@ -128,18 +128,24 @@ void ThemeManager::applyCurrentlySelectedTheme()
 {
     setIconTheme(APPLICATION->settings()->get("IconTheme").toString());
     themeDebugLog() << "<> Icon theme set.";
-    setApplicationTheme(APPLICATION->settings()->get("ApplicationTheme").toString(), true);
+    setApplicationTheme(APPLICATION->settings()->get("ApplicationTheme").toString());
     themeDebugLog() << "<> Application theme set.";
 }
 
-void ThemeManager::setApplicationTheme(const QString& name, bool initial)
+void ThemeManager::setApplicationTheme(const QString& name)
 {
     auto systemPalette = qApp->palette();
     auto themeIter = m_themes.find(name);
     if (themeIter != m_themes.end()) {
         auto& theme = themeIter->second;
         themeDebugLog() << "applying theme" << theme->name();
-        theme->apply(initial);
+
+        // applying the system theme initially seems to cause issues, so lets avoid that
+        if (!m_firstThemeInitialized) {
+            m_firstThemeInitialized = true;
+            return;
+        }

```

Comment:
this check makes it so that my set theme gets ignored, always defaulting to the system theme when opening Prism (i.e. my theme is set to dark, but when opening Prism it defaults to the system theme instead).

Reply:
I had this because the original code had some weird initial theme logic, with that same comment you see here, but i think that is actually completely unnecessary now that i think about it. It might have been added by Peterix back then for #600 🤔

Reply:
removed it 👍